pylint/checkers/utils.py

```diff
@@ -1066,8 +1066,15 @@ def _supports_protocol_method(value: nodes.NodeNG, attr: str) -> bool:
         return False
 
     first = attributes[0]
+
+    # Return False if a constant is assigned
     if isinstance(first, nodes.AssignName):
-        if isinstance(first.parent.value, nodes.Const):
+        this_assign_parent = get_node_first_ancestor_of_type(first, nodes.Assign)
+        assert this_assign_parent is not None
```

Comment:
Are we sure that ``this_assign_parent `` is never None ? What are the downsides of handling the None case ?

Reply:
Thanks, I did miss the sneaky walrus operator. (Turned into a hilarious test case.) So it's worth adding a pragma for the rest. I looked into testing `mixins.AssignTypeMixin` instead, but that pulled in too many classes, e.g. `AssignName` itself.